offchain-modules/packages/app-multisign-server/src/ethMint.ts

```diff
@@ -28,14 +35,45 @@ class EthMint {
       );
     }
 
+    if (!verifyCollector(params)) {
+      return SigResponse.fromSigError(SigErrorCode.InvalidCollector);
+    }
+
     if (await ForceBridgeCore.getXChainHandler().eth!.checkBlockSync!()) {
       return SigResponse.fromSigError(SigErrorCode.BlockSyncUncompleted);
     }
 
     const payload = params.payload as ethMintCollectSignaturesPayload;
 
+    try {
+      await this.verifyRecord(payload.mintRecords);
+    } catch (e) {
+      return new SigResponse(e as SigError);
+    }
+
+    if (!(await this.verifyDuplicated(payload.mintRecords))) {
+      return SigResponse.fromSigError(SigErrorCode.TxCompleted);
+    }
+
     const signature = await this.sign(payload.tx, privateKey);
 
+    await this.signedDb.createSigned(
+      payload.mintRecords.map((record) => {
+        return {
+          sigType: 'mint',
```

Comment:
Conflict with ckbMint: https://github.com/nervosnetwork/force-bridge/blob/3ed65f979a5a2d8041355e91eefb67a64315b4e5/offchain-modules/packages/app-multisign-server/src/ckbSigner.ts#L617

```suggestion
          sigType: 'ethMint',
```